components/DesktopMenu/DesktopMenu.tsx

```diff
@@ -1,20 +1,44 @@
+import * as React from "react"
 import { ChevronUpIcon, ChevronDownIcon } from "@heroicons/react/24/solid"
 import Image from "next/image"
 import { useState } from "react"
 import CustomConnectWallet from "../CustomConnectWallet"
+import { ToggleButton } from "../../shared/Button"
+import { useTheme } from "../../providers/ThemeProvider"
 
 const DesktopMenu = () => {
+  const { onChangeThemeConfig, themeMode } = useTheme()
+
   const [isMenuOpen, setIsMenuOpen] = useState(false)
   const toggleMenu = () => setIsMenuOpen(!isMenuOpen)
+  const [toggle, setToggle] = useState(false)
+
+  const onToggle = () => {
+    setToggle(!toggle)
+    onChangeThemeConfig()
+  }
+
+  React.useEffect(() => {
+    if (themeMode === "light") setToggle(false)
+    if (themeMode === "dark") setToggle(true)
+  }, [themeMode])
```

Comment:
```suggestion
  useEffect(() => {
  setToggle(themeMode !== "light")
  }, [themeMode])
```

Reply:
this accomplishes the same goal right?

Reply:
If users refresh pages, we should set theme mode from local storage variable.
So, I have integrated this part.